File: Cxx/Widgets/CaptionWidget.cxx
#include <vtkPolyDataMapper.h>
#include <vtkActor.h>
#include <vtkRenderWindow.h>
#include <vtkRenderer.h>
#include <vtkRenderWindowInteractor.h>
#include <vtkPolyData.h>
#include <vtkSmartPointer.h>
#include <vtkSphereSource.h>
#include <vtkCaptionWidget.h>
#include <vtkCaptionRepresentation.h>
#include <vtkCaptionActor2D.h>
#include <vtkTextActor.h>
#include <vtkTextProperty.h>

int main(int, char *[])
{
  // Sphere
  vtkSmartPointer<vtkSphereSource> sphereSource = 
    vtkSmartPointer<vtkSphereSource>::New();
  sphereSource->Update();
    
  vtkSmartPointer<vtkPolyDataMapper> mapper = 
    vtkSmartPointer<vtkPolyDataMapper>::New();
  mapper->SetInputConnection(sphereSource->GetOutputPort());

  vtkSmartPointer<vtkActor> actor = 
    vtkSmartPointer<vtkActor>::New();
  actor->SetMapper(mapper);
  
  // A renderer and render window
  vtkSmartPointer<vtkRenderer> renderer = 
    vtkSmartPointer<vtkRenderer>::New();
  vtkSmartPointer<vtkRenderWindow> renderWindow = 
    vtkSmartPointer<vtkRenderWindow>::New();
  renderWindow->AddRenderer(renderer);

  // An interactor
  vtkSmartPointer<vtkRenderWindowInteractor> renderWindowInteractor = 
    vtkSmartPointer<vtkRenderWindowInteractor>::New();
  renderWindowInteractor->SetRenderWindow(renderWindow);
  
  // Create the widget and its representation
  vtkSmartPointer<vtkCaptionRepresentation> captionRepresentation = 
    vtkSmartPointer<vtkCaptionRepresentation>::New();
  captionRepresentation->GetCaptionActor2D()->SetCaption("Test caption");
  captionRepresentation->GetCaptionActor2D()->GetTextActor()->GetTextProperty()->SetFontSize(100);
  
  double pos[3] = {.5,0,0};
  captionRepresentation->SetAnchorPosition(pos);
  
  vtkSmartPointer<vtkCaptionWidget> captionWidget = 
    vtkSmartPointer<vtkCaptionWidget>::New();
  captionWidget->SetInteractor(renderWindowInteractor);
  captionWidget->SetRepresentation(captionRepresentation);
  
  // Add the actors to the scene
  renderer->AddActor(actor);
  renderer->SetBackground(1,0,0);

  renderWindow->Render();
  captionWidget->On();
  
  // Begin mouse interaction
  renderWindowInteractor->Start();
  
  return EXIT_SUCCESS;
}

File: src/Java/Widgets/CaptionWidget.java
import vtk.vtkNativeLibrary;
import vtk.vtkRenderWindow;
import vtk.vtkRenderWindowInteractor;
import vtk.vtkRenderer;
import vtk.vtkActor;
import vtk.vtkNamedColors;
import vtk.vtkPolyDataMapper;
import vtk.vtkSphereSource;
import vtk.vtkCaptionRepresentation;
import vtk.vtkCaptionWidget;

public class CaptionWidget 
{
  // -----------------------------------------------------------------
  // Load VTK library and print which library was not properly loaded
  static 
  {
    if (!vtkNativeLibrary.LoadAllNativeLibraries()) 
    {
      for (vtkNativeLibrary lib : vtkNativeLibrary.values()) 
      {
        if (!lib.IsLoaded()) 
        {
          System.out.println(lib.GetLibraryName() + " not loaded");
        }
      }
    }
    vtkNativeLibrary.DisableOutputWindow(null);
  }
  // -----------------------------------------------------------------

  public static void main(String args[]) 
  {
    vtkNamedColors Color = new vtkNamedColors(); 

    //Renderer Background Color
    double BgColor[] = new double[4];
         
    //Change Color Name to Use your own Color for Renderer Background
    Color.GetColor("red",BgColor);
    
    //Sphere
    vtkSphereSource sphereSource = new vtkSphereSource();
    sphereSource.Update();
    
    //Create a mapper and actor
    vtkPolyDataMapper mapper = new vtkPolyDataMapper();
    mapper.SetInputConnection(sphereSource.GetOutputPort());
    
    vtkActor actor = new vtkActor();
    actor.SetMapper(mapper);
    
    //Create the renderer, render window and interactor.
    vtkRenderer ren = new vtkRenderer();
    vtkRenderWindow renWin = new vtkRenderWindow();
    renWin.AddRenderer(ren);
    vtkRenderWindowInteractor iren = new vtkRenderWindowInteractor();
    iren.SetRenderWindow(renWin);
    
    //Create the widget and its representation
    vtkCaptionRepresentation captionRepresentation = new vtkCaptionRepresentation();
    captionRepresentation.GetCaptionActor2D().SetCaption("Test Caption");
    captionRepresentation.GetCaptionActor2D().GetTextActor().GetTextProperty().SetFontSize(100);

    double pos[] = new double[] {0.5, 0, 0};
    captionRepresentation.SetAnchorPosition(pos);

    vtkCaptionWidget captionWidget = new vtkCaptionWidget();
    captionWidget.SetInteractor(iren);
    captionWidget.SetRepresentation(captionRepresentation);
        
    ren.AddActor(actor);
    ren.SetBackground(BgColor);
    
    renWin.SetSize(300,300);
    renWin.Render();
    captionWidget.On();

    iren.Initialize();
    iren.Start();
  }
}
